frontend/src/components/RequestDialog.vue

```diff
@@ -117,14 +134,22 @@ const {
   error: requestSignerError,
 } = useRequestSigner();
 
-const runRequestSigner = () => {
+const connectMetamask = async () => {
   // TOOD: In future we will not separate getting provider and signer which
   // resolve the undefined provider case.
+  provider.value = await createMetaMaskProvider();
```

Comment:
I think we should show some feedback to the user if there is no MetaMask installed. But maybe that's already going to be addressed with the proper UI.

Also the button will not block / show the spinner until the requestSigner method is called. I think for async calls, it should block from the beginning.

Reply:
> I think we should show some feedback to the user if there is no MetaMask installed. But maybe that's already going to be addressed with the proper UI.
> 
I think this should be handled like Gnosis Safe app as well. We don't show the user anything if no Metamask on his/her browser, and if the user has decided to connect, we redirect him/her to the download page of Metamask!

> Also the button will not block / show the spinner until the requestSigner method is called. I think for async calls, it should block from the beginning.
Yup, nice caught.. I've seen it but totally forgot to fix it.. Thank you!

Reply:
The spinner issue has been fixed!